web/docs/database/operations/actions.md

```diff
@@ -0,0 +1,889 @@
+
+Actions and Queries work hand-in-hand on keeping data caches fresh.
+
+:::tip
+Actions are almost identical to Queries in terms of their API.
+Therefore, if you're already familiar with Queries, you might find reading the entire guide repetitive.
+
+We instead recommend skipping ahead and only reading [the differences between Queries and Actions](#differences-between-queries-and-actions), and consulting the [Options Reference](#options-reference) as needed.
+:::
+
+## Working with Actions
+
+Actions are declared in Wasp and implemented in NodeJS. Wasp executes Actions within the server's context but also generates the code that lets you call them from anywhere in your code (client or server) via the same interface.
+In other words, you don't have to worry about building an HTTP API for the Action, handling the request on the server, or even handling and caching the responses on the client.
+Instead, simply focus on the business logic inside your Action and let Wasp take care of the rest!
+
+To create an Action, you must:
+
+1. Declare the Action in Wasp using the `action` declaration.
+2. Define the Action's NodeJS implementation.
+
+After completing these two steps, you'll be able to use the Action from any point in your code.
+
+### Declaring Actions
+
+We'll start by telling Wasp we wish to create an Action using an `action` declaration.
+Let's declare two Actions, one for creating a task and another one for marking tasks as done:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+action createTask {
+  fn: import { createTask } from "@server/actions.js"
+}
+
+action markTaskAsDone {
+  fn: import { markTaskAsDone } from "@server/actions.js"
+}
+
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+action createTask {
+  fn: import { createTask } from "@server/actions.js"
+}
+
+action markTaskAsDone {
+  fn: import { markTaskAsDone } from "@server/actions.js"
+}
+```
+
+:::warning
+Even though you are using TypeScript and plan to implement this Action in `src/server/actions.ts`, you still need to import it using a `.js` extension. Wasp internally uses `esnext` module resolution, which requires importing all files with a `.js` extension. This is only needed when importing `@server` files.
+
+Read more about ES modules in TypeScript [here](https://www.typescriptlang.org/docs/handbook/esm-node.html). If you're interested in the discussion and the reasoning behind this, read about it [in this GitHub issue](https://github.com/microsoft/TypeScript/issues/33588).
+:::
+
+</TabItem>
+</Tabs>
+
+<small>
+
+If you want to know about all supported options for the `action` declaration, take a look at the [Options reference](#options-reference).
+
+</small>
+
+Wasp Actions and their implementations don't need to (but can) have the same name. We'll keep the names identical to avoid confusion.
```

Comment:
```suggestion
The names of Wasp Actions and their implementations don't necessarily have to match. However, to avoid confusion, we'll keep them the same.
```